es-ds-components/components/es-toggle.vue

```diff
@@ -0,0 +1,133 @@
+<script setup lang="ts">
+import { SwitchRoot, SwitchThumb } from 'reka-ui';
+
+const switchId = useId();
+const model = defineModel<boolean>({
+    default: false,
+});
+
+defineProps({
+    disabled: {
+        type: Boolean,
+        default: false,
+        description: 'Disables the toggle.',
+    },
```

Comment:
Totally optional since this props definition works and we do use this more options-API-like syntax in a lot of ESDS components still, but an alternative that's more TypeScript-y (and potentially more concise when there are a lot of props) is below. The presence or absence of the `?` indicates whether or not it's required, and you can use an actual TypeScript interface to define the types of the props.

```
interface Props {
    disabled?: boolean;
}
withDefaults(defineProps<Props>(), {
    disabled: false,
});
```